packages/wp-source/src/libraries/__tests__/redirections.tests.ts

```diff
@@ -56,7 +44,68 @@ describe("getMatch", () => {
       "/category/nature/forests/"
     );
     expect(redirect("/category/nature", redirections)).toBe("");
-    expect(redirect("/blog/", redirections)).toBe("/");
-    expect(redirect("/blog/some-post/", redirections)).toBe("/some-post/");
+  });
+
+  test("executes redirections in priority order", () => {
+    // Some redirections
+    const redirections: {
+      name: string;
+      priority: number;
+      pattern: string;
+      func: (params: Record<string, string>) => string;
+    }[] = [
+      {
+        name: "",
+        priority: 0,
+        pattern: "/:mainpath+/amp/",
+        func: ({ mainpath }) => `/${mainpath}/`
+      },
+      {
+        name: "",
+        priority: 0,
+        pattern: "/(.*)",
+        func: () => ""
+      },
+      {
+        name: "",
+        priority: 10,
+        pattern: "/subdir/:subpath*",
+        func: ({ subpath }) => (subpath ? `/${subpath}/` : "/")
+      },
+      {
+        name: "",
+        priority: 10,
+        pattern: "/(.*)",
```

Comment:
And another here 👆

Reply:
Ok, i'm going to add a description to the PR and explain there why,